### PR TITLE
Overwrite language switcher

### PR DESCRIPTION
**Pull request by katzmo**

### CODE REVIEW

File: template.php [hunk 1]
```diff
@@ -169,6 +199,15 @@ function campaignion_foundation_block_view_alter(&$data, $block) {
       $link['attributes']['class'][] = $link['attributes']['data-share'] . '-icon';
     }
   }
+  // Remove disabled links from campaignion language switcher blocks.
+  if ($block->module == 'campaignion_language_switcher') {
```
Comment (torotil) [inline template.php]: I think this should be possible in `campaignion_foundation_preprocess_campaignion_language_switcher()` as well.

Comment (katzmo) [reply template.php]: Unfortunately it is not, the links all still have a `href` attribute it the preprocess function, that gets only later removed by `hook_block_view()`.

Comment (torotil) [reply template.php]: Are you sure?

AFAIK:  `campaignion_language_switcher_block_view()` calls `language_negotiation_get_switch_links()` which in turn invokes `hook_language_switch_links_alter()` thereby also `campaignion_language_switcher_language_switch_links_alter()` which removes the `href`. The result of `campaignion_language_switcher_block_view()` is a renderable array so the theme function is not removed until after the function has returned. So the `href` should already be removed when the theme-layer becomes active.

Comment (katzmo) [reply template.php]: Yes. After `campaignion_language_switcher_block_view()` we have the final renderable array as you described, that’s why `campaignion_foundation_block_view_alter()` works. The theme function does its thing before any of that, not after as I would have expected too.

Comment (torotil) [reply template.php]: That would be crazy. Are you sure? It would mean that `theme()` is run before `render()` is run on the same data structure.

Comment (katzmo) [reply template.php]: It looks that way.

**Review by torotil — APPROVED** [review]
Minor thing: The comment to `campaignion_foundation_js_alter()` is not entirely accurate, but never mind.

**Pull request merged**